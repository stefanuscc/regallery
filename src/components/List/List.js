import styles from './List.module.scss'

import ListSkel from './Skeleton/ListSkel'
import ListItem from './Item/ListItem'

export const List = (props) => {
    const { images, isLoading } = props

    return (
        <div className={styles.list}>
            <ul>   
                { isLoading || images == null ? [...Array(10)].map((e, i) => (
                    <ListSkel key={i} idx={i}/>
                )) : ''}
                { isLoading || images == null ? '' : images.map((image, idx) => (
                    <ListItem key={image.id} image={image} idx={idx}/>
                ))}
            </ul>
        </div>
    )
}

export default List