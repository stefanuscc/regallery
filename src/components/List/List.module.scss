// Vars
@import "/src/_vars";

.list {
    width: 100%;
    padding-bottom: 2rem;

    ul {
        list-style: none;
        margin: 0;
        padding: 0;
    }
}