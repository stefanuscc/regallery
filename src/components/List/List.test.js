import React from 'react';
import renderer from 'react-test-renderer';
import List from './List';
import "../../setupTests"

const images = [
    {
        "id": "1016",
        "author": "Philippe Wuyts",
        "width": 3844,
        "height": 2563,
        "url": "https://unsplash.com/photos/_h7aBovKia4",
        "download_url": "https://picsum.photos/id/1016/367/267",
        "name": "Image #1016"
    },
    {
        "id": "1018",
        "author": "Andrew Ridley",
        "width": 3914,
        "height": 2935,
        "url": "https://unsplash.com/photos/Kt5hRENuotI",
        "download_url": "https://picsum.photos/id/1018/367/267",
        "name": "Image #1018"
    },
    {
        "id": "1019",
        "author": "Patrick Fore",
        "width": 5472,
        "height": 3648,
        "url": "https://unsplash.com/photos/V6s1cmE39XM",
        "download_url": "https://picsum.photos/id/1019/367/267",
        "name": "Image #1019"
    }
];

test('List to match snapshot', () => {
    const component = renderer.create(
        <List selected={1016} onClick={() => {}} isLoading={false} images={images}/>
    );
    let tree = component.toJSON();
    expect(tree).toMatchSnapshot();
});