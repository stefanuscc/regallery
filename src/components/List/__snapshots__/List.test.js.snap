// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`List to match snapshot 1`] = `
<div
  className="list"
>
  <ul>
    
    <li
      data-idx={0}
      data-selected={false}
    >
      <div
        className="item"
      >
        <div
          className="container"
        >
          <div
            className="ibox"
          >
            <img
              alt=""
              className="image"
            />
          </div>
        </div>
        <div
          className="attributes"
        >
          <div>
            <p
              className="title"
            >
              Author
            </p>
            <p
              className="name"
            >
              <span
                className="truncate"
              >
                Philippe Wuyts
              </span>
            </p>
          </div>
          <div>
            <p
              className="title"
            >
              Resolutions
            </p>
            <p
              className="name"
            >
              <span
                className="truncate"
              >
                3844
                 x 
                2563
              </span>
            </p>
          </div>
          <div>
            <p
              className="title"
            >
              Image Name
            </p>
            <p
              className="name"
            >
              Image #1016
            </p>
          </div>
        </div>
        <svg
          className="button"
          fill="none"
          onClick={[Function]}
          stroke="currentColor"
          strokeWidth="2"
          viewBox="0 0 24 24"
        >
          <path
            d="M15.232 5.232l3.536 3.536m-2.036-5.036a2.5 2.5 0 113.536 3.536L6.5 21.036H3v-3.572L16.732 3.732z"
            strokeLinecap="round"
            strokeLinejoin="round"
          />
        </svg>
      </div>
    </li>
    <li
      data-idx={1}
      data-selected={false}
    >
      <div
        className="item"
      >
        <div
          className="container"
        >
          <div
            className="ibox"
          >
            <img
              alt=""
              className="image"
            />
          </div>
        </div>
        <div
          className="attributes"
        >
          <div>
            <p
              className="title"
            >
              Author
            </p>
            <p
              className="name"
            >
              <span
                className="truncate"
              >
                Andrew Ridley
              </span>
            </p>
          </div>
          <div>
            <p
              className="title"
            >
              Resolutions
            </p>
            <p
              className="name"
            >
              <span
                className="truncate"
              >
                3914
                 x 
                2935
              </span>
            </p>
          </div>
          <div>
            <p
              className="title"
            >
              Image Name
            </p>
            <p
              className="name"
            >
              Image #1018
            </p>
          </div>
        </div>
        <svg
          className="button"
          fill="none"
          onClick={[Function]}
          stroke="currentColor"
          strokeWidth="2"
          viewBox="0 0 24 24"
        >
          <path
            d="M15.232 5.232l3.536 3.536m-2.036-5.036a2.5 2.5 0 113.536 3.536L6.5 21.036H3v-3.572L16.732 3.732z"
            strokeLinecap="round"
            strokeLinejoin="round"
          />
        </svg>
      </div>
    </li>
    <li
      data-idx={2}
      data-selected={false}
    >
      <div
        className="item"
      >
        <div
          className="container"
        >
          <div
            className="ibox"
          >
            <img
              alt=""
              className="image"
            />
          </div>
        </div>
        <div
          className="attributes"
        >
          <div>
            <p
              className="title"
            >
              Author
            </p>
            <p
              className="name"
            >
              <span
                className="truncate"
              >
                Patrick Fore
              </span>
            </p>
          </div>
          <div>
            <p
              className="title"
            >
              Resolutions
            </p>
            <p
              className="name"
            >
              <span
                className="truncate"
              >
                5472
                 x 
                3648
              </span>
            </p>
          </div>
          <div>
            <p
              className="title"
            >
              Image Name
            </p>
            <p
              className="name"
            >
              Image #1019
            </p>
          </div>
        </div>
        <svg
          className="button"
          fill="none"
          onClick={[Function]}
          stroke="currentColor"
          strokeWidth="2"
          viewBox="0 0 24 24"
        >
          <path
            d="M15.232 5.232l3.536 3.536m-2.036-5.036a2.5 2.5 0 113.536 3.536L6.5 21.036H3v-3.572L16.732 3.732z"
            strokeLinecap="round"
            strokeLinejoin="round"
          />
        </svg>
      </div>
    </li>
  </ul>
</div>
`;
